cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -154,33 +158,60 @@
-   * @return this {@link ScheduleBuilder}
+   * @return a {@link ScheduleCreationSpec}
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(String programNamespace, String application, String appVersion,
-                                                 ProgramType programType, String program,
-                                                 ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, String application, String appVersion,
+                                              ProgramType programType, String program,
+                                              ProgramStatus... programStatuses);
 
   /**
    * Creates a schedule which is triggered in the same application version.
    *
-   * @see ScheduleBuilder#triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(String programNamespace, String application, ProgramType programType,
-                                                 String program, ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, String application, ProgramType programType,
+                                              String program, ProgramStatus... programStatuses);
 
   /**
    * Creates a schedule which is triggered in the same application and application version.
    *
-   * @see ScheduleBuilder#triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(String programNamespace, ProgramType programType, String program,
-                                                 ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, ProgramType programType, String program,
+                                              ProgramStatus... programStatuses);
 
   /**
    * Creates a schedule which is triggered in the same namespace, application, and application version.
    *
-   * @see ScheduleBuilder#triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(ProgramType programType, String program,
-                                                 ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(ProgramType programType, String program,
+                                              ProgramStatus... programStatuses);
+
+  /**
+   * Inner class that creates a ScheduleCreationSpec from the deployed properties
+   */
+  class ScheduleCreationBuilder extends ScheduleCreationSpec {
```

Comment:
`DefaultScheduleBuilder` needs to be able to access it though since `triggerOnProgramStatus` needs to instantiate `ScheduleCreationBuilder`.

If it were a private inner class of `DefaultAppConfigurer`, we would have to do something like:

```
return new DefaultAppConfigurer(namespace, artifactId, app).new ScheduleCreationBuilder(...)
```